### DISCUSSION TITLE
Is there any performance difference? ISR vs SSR with Stale-While-Revalidate Headers

### DISCUSSION THREAD

**Discussion by darrylmorley** [vercel/next.js#25703] [Help]
I may have the wrong end of the stick but does SSR with the stale-while-revalidate header basically give you the same benefits of using ISR?

It appears at first glance to initially be SSR and then change to ISR, is that the correct understanding?

**Comment by richardscarrott** [comment]
@darrylmorley this is how I understand it; I can't see the purpose of adding SSG / ISR over SSR + caching on a site as it seems like a whole lot of complexity just to improve the very first visit to a page?
